src/libopensc/card-piv.c

```diff
@@ -2780,8 +2794,11 @@ static int piv_generate_key(sc_card_t *card,
 				keydata->pubkey_len = taglen;
 				memcpy (keydata->pubkey, tag, taglen);
 			}
-		}
-		else { /* must be EC */
+			//		} else if (keydata->key_algid == 0xE0 || keydata->key_algid == 0xE1) {
+			//			/* TODO DEE need to look at what gets returned */
+			/* TODO assume same as EC with tag 86 */
+
```

Comment:
Only Yubikey  version > 5.7.1 support  RSA 4096, Ed25519 and X25519 as a new feature, and I have not finished testing it and these are notes to myself in the code. I will hold off on changing this for now.

 If it would help get this PR accepted, I can pull all the PIV  RSA 4096, Ed25519 and X25519 code, and make it a separate PR.

Reply:
Removing comments and adding error checks. The same tags are used when creating a EC, ED25519 and X25519 keys.